packages/snippet/src/UserInterface/Controller/Admin/SnippetAreaController.php

```diff
@@ -0,0 +1,129 @@
+        MessageBusInterface $messageBus,
+        private NormalizerInterface $normalizer,
+        private FieldDescriptorFactoryInterface $fieldDescriptorFactory,
+        private DoctrineListBuilderFactoryInterface $listBuilderFactory,
+        private RestHelperInterface $restHelper,
+        private SnippetAreaRepositoryInterface $snippetAreaRepository,
+        private array $snippetArea,
+    ) {
+        // Setting the message bus of the HandleTrait
+        $this->messageBus = $messageBus;
+    }
+
+    public function cgetAction(Request $request): Response
+    {
+        /** @var DoctrineFieldDescriptorInterface[]|null $fieldDescriptors */
+        //$fieldDescriptors = $this->fieldDescriptorFactory->getFieldDescriptors(SnippetAreaInterface::RESOURCE_KEY);
+        //Assert::notNull($fieldDescriptors, 'Could not find field descriptors for resource key: ' . SnippetAreaInterface::RESOURCE_KEY);
+
+        /** @var DoctrineListBuilder $listBuilder */
+        //$listBuilder = $this->listBuilderFactory->create(SnippetAreaInterface::class);
+        //$listBuilder->setIdField($fieldDescriptors['id']); // We need to set this because it's the uuid doctrine column
+        //$listBuilder->setParameter('locale', $request->query->get('locale'));
+        //$listBuilder->setParameter('webspace', $request->query->get('webspace'));
+
+        //$this->restHelper->initializeListBuilder($listBuilder, $fieldDescriptors);
+
+        $snippetAreas = $this->snippetAreaRepository->findByWebspace($request->query->get('webspace'));
+
+        // Add the empty snippet areas as placeholders
+        foreach ($this->snippetArea as $snippetArea) {
+            $key = $snippetArea['key'];
+            if (!\array_key_exists($key, $snippetAreas)) {
+                $snippetAreas[$key] = new SnippetArea(
+                    areaKey: $snippetArea['key'],
+                    webspaceKey: $request->attributes->getString('webspace'),
+                );
+            }
+        }
+
+        $listRepresentation = new CollectionRepresentation(
+            \array_values($snippetAreas),
+            SnippetAreaInterface::RESOURCE_KEY,
+        );
+
+        return new JsonResponse($this->normalizer->normalize(
+            $listRepresentation->toArray(),
+            'json',
+            [
+                'locale' => $request->getLocale(),
+                'sulu_admin' => true,
+                'sulu_admin_snippet' => true,
+                'sulu_admin_snippet_list' => true,
+            ],
+        ));
+    }
+
+    public function putAction(Request $request): Response
+    {
+        $message = new ModifySnippetAreaMessage([
+            ...$request->attributes->all('_route_params'),
+            ...$request->request->all(),
+        ]);
+
+        /** @see Sulu\Snippet\Application\MessageHandler\SetSnippetAreaMessageHandler */
+        $this->handle(new Envelope($message, [new EnableFlushStamp()]));
+
+        return new Response(null, Response::HTTP_OK);
```

Comment:
We should still return the snippetArea values here, to have the API similar to the old one